src/_arraykit.c

```diff
@@ -383,7 +383,35 @@ dtype_from_element(PyObject *Py_UNUSED(m), PyObject *arg)
 
     // Integers
     if (PyLong_CheckExact(arg)) {
-        return (PyObject*)PyArray_DescrFromType(NPY_LONG);
+        // This move cuts our speed gain from ~2.5 to ~2.0 :(
+        PyObject *scalar = PyArray_FromAny(arg, NULL, 0, 0, 0, NULL);
+        if (!scalar) {
+            return NULL;
+        }
+        return (PyObject*)PyArray_DESCR((PyArrayObject*)scalar);
```

Comment:
This leaks a reference to `scalar` (from which the returned dtype is borrowed). Maybe something like:

```suggestion
        PyObject *dtype = (PyObject*)PyArray_DESCR((PyArrayObject*)scalar);
        Py_XINCREF(dtype);
        Py_DECREF(scalar);
        return dtype;
```

Have you checked out [`PyArray_DescrFromObject`](https://numpy.org/doc/stable/reference/c-api/array.html#c.PyArray_DescrFromObject), though? It looks like exactly what you need.

Reply:
They have a lot of apis. I looked through a lot of them to find the best one, but I must have missed this. It's definitely better.